### PR TITLE
Django_HW7

### PR DESCRIPTION
**Pull request by palagian**

### CODE REVIEW

**Comment by palagian** [comment]
Can't get why requirements.txt pulls empty.
There is data:
alembic==1.11.1
amqp==5.1.1
asgiref==3.7.2
async-timeout==4.0.2
billiard==4.1.0
blinker==1.6.2
celery==5.3.1
certifi==2022.12.7
charset-normalizer==3.1.0
click==8.1.3
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.3.0
colorama==0.4.6
cron-descriptor==1.4.0
Django==4.2.3
django-celery-beat==2.5.0
django-filter==23.2
django-timezone-field==5.1
djangorestframework==3.14.0
Flask==2.3.2
Flask-Migrate==4.0.4
Flask-Script==2.0.6
flask-sqlalchemy==3.0.5
greenlet==2.0.2
idna==3.4
itsdangerous==2.1.2
Jinja2==3.1.2
kombu==5.3.1
Mako==1.2.4
MarkupSafe==2.1.3
prompt-toolkit==3.0.39
python-crontab==3.0.0
python-dateutil==2.8.2
python-dotenv==1.0.0
pytz==2023.3
redis==4.6.0
requests==2.29.0
six==1.16.0
SQLAlchemy==2.0.19
sqlparse==0.4.4
typing-extensions==4.7.1
tzdata==2023.3
urllib3==1.26.15
vine==5.0.0
wcwidth==0.2.6
Werkzeug==2.3.6

File: Django_HW/users/tasks.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@shared_task
+def print_hello_celery():
+    print("This is my first celery task")
+
+
+@shared_task
+def print_users_purchases(user_id):
+    try:
+        user = User.objects.get(pk=user_id)
+        purchase_count = Purchase.objects.filter(user=user).count()
```
Comment (acman) [inline Django_HW/users/tasks.py:16]: nit: could be filtered by user__id=user_id or even user=user_id

**Review by acman — CHANGES_REQUESTED** [review]
Don't see usage of print_purchase_count

**Comment by palagian** [comment]
@acman please explain what do you mean.
In HW task was no task for creating print_purchase_count. 
Only count of purchases by users: print_users_purchases.

Should I create additional @shared_task for counting total purchases?
Thank you!

**Comment by acman** [comment]
Oh, sorry. I meant `print_users_purchases`. It is fine that tasks were created but they should be used somewhere in views

File: Django_HW/users/views.py [hunk 2]
```diff
@@ -11,3 +13,11 @@ class UserDetailView(DetailView):
     model = User
     template_name = 'user_detail.html'
     context_object_name = 'user'
+
+
+def user_list(request):
+    users = User.objects.all()
+    user_ids = [user.id for user in users]
+    print_users_purchases.delay(user_ids)
```
Comment (acman) [inline Django_HW/users/views.py:21]: 👍 it could be used inside any class method,  e.g. `get_queryset`

**Review by acman — APPROVED** [review]
GJ!

**Pull request merged**